packages/amplify_flutter/lib/config/analytics/analytics_config.dart

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_flutter/config/analytics/analytics_plugin.dart';
+
+part 'analytics_config.g.dart';
+
+/// A Dart representation of an Amplify Analytics configuration
+class AnalyticsConfig {
```

Comment:
So sorry for the confusion - the `@JsonSerializable` annotation is needed on all class requiring codegen, but individual fields only need the `@JsonKey` annotation if they are overriding options from `JsonSerializable` or need to recase/rename the generated JSON key.

Reply:
This should be resolved.